atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java

```diff
@@ -37,41 +37,124 @@
 
 /**
  * Provides the methods for a transaction with the key-value store.
+ *
+ * In general: users may assume that if maps (including sorted maps) keyed on byte[] are returned to the user,
+ * they may be accessed, including via random access, via any byte array that is equivalent in terms of
+ * {@link java.util.Arrays#equals(byte[], byte[])}.
+ *
  * @see TransactionManager
  */
 public interface Transaction {
 
+    /**
```

Comment:
and, just to make sure... do other Transaction implementations such as CachedTransaction obey these constraints? I know we have AbstractTransactionTests, do those get run on CachedTransactions? Personally I'd be minded to write a 'contract validating transaction' which we can wrap our transactions in.

Reply:
Is AbstractTransactionTest run on CachingTransactions: good flag. Not previously, but now it is - we've tested for almost all of the contractual nuances by now, and it passed all of the tests.

Reply:
Most of the others just delegate (or in the case of ReadTransaction delegate the relevant things), from a quick pass.

Reply:
(Turns out there are some weird CircleCI issues that I can't reproduce locally around TMs being shut. I wouldn't block on this currently)